src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -199,11 +199,20 @@ public void FailRequest(Exception error)
 
             private void SetUrl()
             {
-                EventSourceTrace("Url: {0}", _requestMessage.RequestUri);
-                SetCurlOption(CURLoption.CURLOPT_URL, _requestMessage.RequestUri.AbsoluteUri);
+                string url = GetAbsoluteUri(_requestMessage.RequestUri);
+                EventSourceTrace("Url: {0}", url);
+                SetCurlOption(CURLoption.CURLOPT_URL, url);
                 SetCurlOption(CURLoption.CURLOPT_PROTOCOLS, (long)(CurlProtocols.CURLPROTO_HTTP | CurlProtocols.CURLPROTO_HTTPS));
             }
 
+            private static string GetAbsoluteUri(Uri uri)
+            {
+                // uri.AbsoluteUri/ToString() do not include IPv6 scope IDs, but when provided they
+                // should be used or else libcurl may be unable to correctly connect.  SerializationInfoString
+                // ensures these details are included.
+                return uri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
```

Comment:
>I don't see anything on Uri that lets me check whether the host has a scope ID in it.

I don't think there is a method/property to do it explicitly. It might be possible if you extracted the `Host` portion of the Uri and then use that with an `IPAddress` or related class to 1) determine that it is an IPv6 literal and 2) parse out via the APIs the scope portion.